gb-cpu/src/cpu.rs

```diff
@@ -10,6 +10,7 @@ pub struct Cpu {
     pub registers: Registers,
     pub controller: MicrocodeController,
     pub io_regs: Rc<RefCell<IORegisters>>,
+    pub halted: bool,
```

Comment:
to be more clear and to not conflict with the cpu state, maybe renambe the variable to something like `halted_dma` ?